packages/pwa-kit-dev/src/configs/webpack/config.js

```diff
@@ -433,7 +439,10 @@ const client =
                 // Must be named "client". See - https://www.npmjs.com/package/webpack-hot-server-middleware#usage
                 name: CLIENT,
                 // use source map to make debugging easier
-                devtool: mode === development ? 'source-map' : false,
+                devtool:
+                    mode === development || process.env.PWA_KIT_SOURCE_MAP === 'true'
```

Comment:
@joeluong-sfcc asked me some questions yesterday about [this](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1535) PR I did in the past to help with dev server crashing. It looks like this change that you have here kinda reverses that work, e.g. it looks like source maps are always included for development, meaning that the dev server might possibly ramp up it's memory footprint faster and crash sooner.

Reply:
@bendvc Can you elaborate how this would reverse the work in your PR. I checked the PR you mentioned, you changed the code for request-processor [here](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1535/files#diff-3c62fd04e584e6cbc2443d75ef447133665c8c21a1b34e0fa8b9d5a29cd88808R547). The source-map has always available for development regardless of there is a new/old variable. 🤔 . I don't think my change will affect the dev server bc the issue is already existing for development regardless of the variable env in this PR.

Reply:
It was [this](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1535/files#diff-3c62fd04e584e6cbc2443d75ef447133665c8c21a1b34e0fa8b9d5a29cd88808R478) line in particular that I was worried about.. it looks like it might have even been changed after my pr.. 🤔

Reply:
oh, I got what you meant.  I believed this is not the right place. You may want to look into this code [line](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2240/files#diff-3c62fd04e584e6cbc2443d75ef447133665c8c21a1b34e0fa8b9d5a29cd88808R502). In your PR, you added INSPECT for two bundles the server and request-processor bundle, I did not remove the inspect flag for either of them. So this PR won't reverse any of yours. I will gonna have a quick check to see if I will have the heap map issue locally.

Reply:
Tested the local server with script in this [PR](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1612) on Node 18
```
for i in $(seq 1 1000); do
  echo "// $i" >> app/constants.js
  sleep 3
  code=`curl -sSI -o /dev/null -w "%{http_code}" http://localhost:3000` || break
  echo "$i: $code"
done
```
I got the server crashed on iteration 63 on Node 18 and 20, and around 62 to Node 22, which is kinda consistent with existing behavior (ranging from 59-63 iteration when crashing). I don't think the change in this PR will contribute more into the memory heap problem